packages/x-components/src/x-modules/search/components/banner.vue

```diff
@@ -1,15 +1,30 @@
 <template>
   <a
+    v-if="banner.url && !imageError"
     @click="emitClickEvent"
-    @click.right="emitClickEvent"
     @click.middle="emitClickEvent"
+    @click.right="emitClickEvent"
     :href="banner.url"
     class="x-banner"
     data-test="banner"
   >
-    <img :src="banner.image" class="x-banner__image" :alt="banner.title" />
-    <h2 class="x-banner__title">{{ banner.title }}</h2>
+    <img
+      @error="imageError = true"
+      :src="banner.image"
+      :alt="banner.title ? banner.title : 'Banner'"
+      class="x-banner__image"
+    />
+    <h2 v-if="banner.title" class="x-banner__title">{{ banner.title }}</h2>
   </a>
+  <figure v-else-if="!imageError" class="x-banner" data-test="banner">
+    <img
+      @error="imageError = true"
+      :src="banner.image"
+      :alt="banner.title ? banner.title : 'Banner'"
+      class="x-banner__image"
+    />
+    <h2 v-if="banner.title" class="x-banner__title">{{ banner.title }}</h2>
```

Comment:
This is duplicated. Try to find a solution that doesn't have this duplicity.

Changing the root element to a conditional `component` element and making adding the events dynamic could work:

````
<component v-if="!imageError" :is="banner.url ? 'a' : 'figure'"
    v-if="!imageError"
    v-on="banner.url ? { // don't set this object here
      click: emitClickEvent,
      auxclick: emitClickEvent,
      contextmenu: emitClickEvent
    } : {}"
    :href="banner.url"
    class="x-banner"
    data-test="banner"
  >
    <img
      @error="imageError = true"
      :src="banner.image"
      :alt="banner.title ? banner.title : 'Banner'"
      class="x-banner__image"
    />
    <h2 v-if="banner.title" class="x-banner__title">{{ banner.title }}</h2>
  </component>
````

Other solution could be moving the click anchor and logic to a component and the do:

````
<component v-if="!imageError" :is="banner.url ? 'newComponent' : 'figure'"
    v-if="!imageError"
    class="x-banner"
    data-test="banner"
  >[...]
````

Reply:
done